tools/failures/detect_new_failures.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python
+# Copyright 2015 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Detect new flakes and create issues for them"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import datetime
+import json
+import logging
+import os
+import pprint
+import sys
+import urllib
+import urllib2
+from collections import namedtuple
+
+gcp_utils_dir = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), '../gcp/utils'))
+sys.path.append(gcp_utils_dir)
+
+import big_query_utils
+
+GH_ISSUE_CREATION_URL = 'https://api.github.com/repos/grpc/grpc/issues'
+GH_ISSUE_SEARCH_URL = 'https://api.github.com/search/issues'
+KOKORO_BASE_URL = 'https://kokoro2.corp.google.com/job/'
+
+def gh(url, data=None):
+  request = urllib2.Request(url, data=data)
+  assert TOKEN
+  request.add_header('Authorization', 'token {}'.format(TOKEN))
+  if data:
+    request.add_header('Content-type', 'application/json')
+  response = urllib2.urlopen(request)
+  if 200 <= response.getcode() < 300:
+    return json.loads(response.read())
+  else:
+    raise ValueError('Error ({}) accessing {}'.format(
+        response.getcode(), response.geturl()))
+
+
+def search_gh_issues(search_term, status='open'):
+  params = ' '.join((search_term, 'is:issue', 'is:open', 'repo:grpc/grpc'))
+  qargs = urllib.urlencode({'q': params})
+  url = '?'.join((GH_ISSUE_SEARCH_URL, qargs))
+  response = gh(url)
+  return response
+
+
+def create_gh_issue(title, body, labels, assignees=[]):
+  params = {'title': title,
+            'body': body,
+            'labels': labels}
+  if assignees:
+    params['assignees'] = assignees
+  data = json.dumps(params)
+  response = gh(GH_ISSUE_CREATION_URL, data)
+  issue_url = response['html_url']
+  print('Created issue {} for {}'.format(issue_url, title))
+
+
+def build_kokoro_url(job_name, build_id):
+  job_path = '{}/{}'.format('/job/'.join(job_name.split('/')), build_id)
+  return KOKORO_BASE_URL + job_path
+
+
+def create_issues(new_flakes, always_create):
+  for test_name, results_row in new_flakes.items():
+    poll_strategy, job_name, build_id, timestamp = results_row
+    url = build_kokoro_url(job_name, build_id)
```

Comment:
nit: add a TODO that kokoro build url has only temporary lifetime and we are only using it because we don't have access to the sponge link